PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -122,13 +131,34 @@ PATMuonProducer::PATMuonProducer(const edm::ParameterSet & iConfig) : useUserDat
 
   //for mini-isolation calculation
   computeMiniIso_ = iConfig.getParameter<bool>("computeMiniIso");
+
   miniIsoParams_ = iConfig.getParameter<std::vector<double> >("miniIsoParams");
   if(computeMiniIso_ && miniIsoParams_.size() != 9){
       throw cms::Exception("ParameterError") << "miniIsoParams must have exactly 9 elements.\n";
   }
   if(computeMiniIso_)
       pcToken_ = consumes<pat::PackedCandidateCollection >(iConfig.getParameter<edm::InputTag>("pfCandsForMiniIso"));
 
+  // standard selectors
+  recomputeBasicSelectors_ = iConfig.getParameter<bool>("recomputeBasicSelectors");
+  computeMuonMVA_ = iConfig.getParameter<bool>("computeMuonMVA");
+  if (computeMuonMVA_ and not computeMiniIso_) 
+    throw cms::Exception("ConfigurationError") << "MiniIso is needed for Muon MVA calculation.\n";
+
+  if (computeMuonMVA_) {
+    // pfCombinedInclusiveSecondaryVertexV2BJetTags
+    mvaBTagCollectionTag_  = consumes<reco::JetTagCollection>(iConfig.getParameter<edm::InputTag>("mvaJetTag"));
+    mvaL1Corrector_        = consumes<reco::JetCorrector>(iConfig.getParameter<edm::InputTag>("mvaL1Corrector"));
+    mvaL1L2L3ResCorrector_ = consumes<reco::JetCorrector>(iConfig.getParameter<edm::InputTag>("mvaL1L2L3ResCorrector"));
+    rho_                   = consumes<double>(iConfig.getParameter<edm::InputTag>("rho"));
+    mvaUseJec_ = iConfig.getParameter<bool>("mvaUseJec");
+    mvaDrMax_  = iConfig.getParameter<double>("mvaDrMax");
+
+    // xml training file
+    edm::FileInPath fip("RecoMuon/MuonIdentification/data/mu_BDTG.weights.xml");
```

Comment:
if this is not going to be fixed and always working/valid for run1, run2, phase-2 , then this parameter should be made configurable so that an appropriate training is picked up.

Reply:
fixed